src/SelfService/Application/InvitationApplicationService.cs

```diff
@@ -149,6 +166,29 @@ Capability capability
                 _logger.LogWarning("Unable to parse invitee \"{Invitee}\" as a valid user id", invitee);
                 continue;
             }
+
+            if (await _membershipRepository.IsAlreadyMember(capability.Id, inviteeId))
+            {
+                _logger.LogWarning(
+                    "User \"{Invitee}\" is already member of \"{Capability}\"",
+                    inviteeId,
+                    capability.Id
+                );
+                continue;
+            }
+            var existingInvitations = await _invitationRepository.GetAllWithPredicate(
+                x => x.Invitee == invitee && x.TargetId == capability.Id && x.Status == InvitationStatusOptions.Active
+            );
```

Comment:
maybe extract this to a function, you are doing the same in AcceptInvitation, but existingInvitations are called similarInvitations.

Reply:
I still like the naming (as it makes sense in the context),  but I have extracted to functions that may have some better names.